src/selection-set-to-js.js

```diff
@@ -38,6 +39,14 @@ function applyArguments(options, selection, operationName, config) {
   options.push(argToJS(t.identifier('args'), selection.arguments, operationName, config));
 }
 
+function applyDirectives(options, selection, operationName, config) {
+  if (!(selection.directives && selection.directives.length)) {
```

Comment:
Accidentally resolved but yes. Selection is the field this is acting on. It sound always be present.

Reply:
Since it's coming from the selectionSet's selection array, it should always be defined given the type definitions from `graphql-js` https://github.com/graphql/graphql-js/blob/8960d0d8d7b251a8031c988b1f70a3e962bd482c/src/language/ast.d.ts#L240